src/zenml/integrations/aws/flavors/sagemaker_orchestrator_flavor.py

```diff
@@ -0,0 +1,90 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Amazon SageMaker orchestrator flavor."""
+
+from typing import TYPE_CHECKING, Optional, Type
+
+from zenml.config.base_settings import BaseSettings
+from zenml.integrations.aws import AWS_SAGEMAKER_STEP_OPERATOR_FLAVOR
+from zenml.orchestrators.base_orchestrator import BaseOrchestratorFlavor
+from zenml.step_operators.base_step_operator import BaseStepOperatorConfig
+
+if TYPE_CHECKING:
+    from zenml.integrations.aws.orchestrators import SagemakerOrchestrator
+
+
+class SagemakerOrchestratorSettings(BaseSettings):
```

Comment:
Here is where my biggest feedback comes in for this implementation. We can provide way more  control to the user on how the pipeline is executed IMO in Settings or config. I have made individual comments below but here is a summary

* Give them the chance to specify the resources per step and translate to an approapriate instance_type
* Give them the chance to attach a GPU to the instance (Same as above)
* Give them the chance to override the role
* Give the chance to pass env variables
* Potentially let them override the `Processor` or `ProcessingStep` (Not sure if possible tbh)
* Explore other options that are 'normal' to override in a sagemaker pipeline and expose them via these settings or config params


Basically, we have to think from the users perspective: What are the constraints they have usually when running pipelines? What regions are accessible to them? what buckets? What sort of control would they like on an infrastructure level?

Reply:
I agree with most of the points by @htahir1 above, but I wouldn't worry about letting them switch from `Processor` for now.

Also needs some docstrings to explain the various attributes.